tools/deploy.py

```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/env python3
+import argparse
+import datetime
+import functools
+import hashlib
+import json
+import os
+import pathlib
+import shutil
+import subprocess
+import sys
+import urllib.parse
+import urllib.request
+
+
+def url_fetch(url, headers=None, **kwargs):
+    """Make a web request to the given URL and return the response object."""
+    request_headers = {
+        # Override the User-Agent because our download server seems to block
+        # requests with the default UA value and responds "403 Forbidden".
+        "User-Agent": (
+            "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:86.0) Gecko/20100101 "
+            "Firefox/86.0"
+        ),
+    }
+    if headers:
+        request_headers.update(headers)
+    req = urllib.request.Request(url, headers=request_headers, **kwargs)
+    return urllib.request.urlopen(req, timeout=10)
+
+
+def url_exists(url):
+    """Make a HEAD request to the URL and check if the response is "200 OK"."""
+    try:
+        resp = url_fetch(url, method="HEAD")
+    except IOError:
+        return False
+    return resp.status == 200
+
+
+def url_download_json(url):
+    """Returns the JSON object from the given URL or return None."""
+    try:
+        resp = url_fetch(url)
+        manifest_data = resp.read().decode()
+    except IOError:
+        return None
+
+    return json.loads(manifest_data)
+
+
+def sha256(file_path):
+    """Returns the sha256 hexdigest for a file."""
+    with open(file_path, mode="rb") as fp:
+        read_chunk = functools.partial(fp.read, 1024)
+        m = hashlib.sha256()
+        for data in iter(read_chunk, b""):
+            m.update(data)
+        return m.hexdigest()
+
+
+def git_info(info, path="."):
+    """Return the checked out git branch for the given path."""
+    if info == "branch":
+        cmd = ("git", "rev-parse", "--abbrev-ref", "HEAD")
+    elif info == "commit":
+        cmd = ("git", "rev-parse", "HEAD")
+    elif info == "describe":
+        cmd = ("git", "describe")
```

Comment:
How about adding this to the manifest.json?

Reply:
We can, but how do you want to display it on the website? Do you want to show it instead of the commit-id?

Reply:
Yes. After #3259 is merged, users will be able to easily compare that to the `git describe` in the application.

Reply:
Ok, I'll take care of that in a follow-up PR. I also wanted to update the package slugs, so it will need changes on the website repo anyway.